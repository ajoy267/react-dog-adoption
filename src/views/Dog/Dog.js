import React from 'react';
import { fetchDogById } from '../../services/dog';
import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
import DogDetails from '../../components/DogDetails/DogDetails';
import './Dog.css';

export default function Dog() {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    let timer;
    const fetchData = async () => {
      const data = await fetchDogById(params.id);
      setDog(data);
      timer = setTimeout(() => {
        setLoading(false);
      }, 750);
    };
    if (loading) {
      fetchData();
    }
    return () => {
      clearInterval(timer);
    };
  }, [params.id, loading]);
  return (
    <>
      {loading && <div className="loader"></div>}
      {!loading && (
        <div>
          <DogDetails dog={dog} />
        </div>
      )}
    </>
  );
}
