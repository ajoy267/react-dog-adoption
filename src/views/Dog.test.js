import { MemoryRouter, Route } from 'react-router-dom/cjs/react-router-dom.min';
import Dog from '../views/Dog/Dog';
import { render, screen } from '@testing-library/react';

test('should render an individual dog', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/10']}>
      <Route path="/dogs/:id">
        <Dog />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('Barton');
  expect(container).toMatchSnapshot();
});
