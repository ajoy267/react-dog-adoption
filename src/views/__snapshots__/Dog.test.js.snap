// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an individual dog 1`] = `
<div>
  <div>
    <div>
      <h1>
        Barton
      </h1>
      <img
        src="https://placedog.net/500?id=6"
      />
      <p>
        Barton
         is a(n) 
        1
         year old 
        Weimaraner
      </p>
      <p>
        Adores every person he meets. Very social and lots of energy
      </p>
    </div>
  </div>
</div>
`;
